java-frontend/src/main/java/org/sonar/java/se/constraint/ObjectConstraint.java

```diff
@@ -75,6 +77,17 @@ public boolean hasStatus(@Nullable Object aState) {
     }
     return aState.equals(status);
   }
+  
+  public static ObjectConstraint constraintWithStatus(ProgramState programState, SymbolicValue value, Object aState) {
```

Comment:
As discussed previously: this methods introduced here put a dependency on ProgramState from ObjectConstraint which is not a good thing.